nomos-libp2p/src/lib.rs

```diff
@@ -108,7 +114,19 @@ impl Swarm {
 
         swarm.listen_on(multiaddr!(Ip4(config.host), Tcp(config.port)))?;
 
-        Ok(Swarm { swarm })
+        #[cfg(feature = "metrics")]
+        let metrics = {
+            let mut metric_registry = Registry::default();
```

Comment:
Is it true, that if we'll want to have one prometheus service inside the node (spawned as described in TODO comment) then `metrics_registry` will need to be passed as reference instead of created here?

Reply:
ah, I need to think about the way how to make this work with the existing metrics service. Let me check.

Reply:
Hm. I think that the way how libp2p gathers/exposes metrics is not flexible. Even if we add a `prometheus` backend in our metrics service, we need to pass the reference of `prometheus_client::registry::Registry` to the libp2p network service, so that it can be used as described in this PR. Also, it seems that the only way we can query metrics from the `Registry` is [dumping all metrics](https://github.com/libp2p/rust-libp2p/blob/c1551d71b6a7ad673706d7106c797fe464cb2820/examples/metrics/src/http_service.rs#L69-L69) using the text encoder or others. I think that this way doesn't fit with the design of our metrics service which has `update()` and `load()` functions. 

I'd like to suggest to postpone this PR since this topic isn't in priority. Later when we really begin using our metrics service, we can think about how we can adopt prometheus-client (or if it's necessary).